plugin/src/patcher.ml4

```diff
@@ -103,42 +107,68 @@ let invert_patch n env patch sigma =
     failwith "Could not find a well-typed inverted term"
 
 (* Common patch command functionality *)
-let patch env n try_invert a search sigma =
+    
+let patch env n try_invert a search sigma define =
   let reduce = try_reduce reduce_remove_identities in
   let sigma, patch_to_red = search env a sigma in
   let sigma, patch = reduce env sigma patch_to_red in
   let prefix = Id.to_string n in
-  ignore (define_term n sigma patch false);
   (if !opt_printpatches then
     print_patch env sigma prefix patch
   else
     Printf.printf "Defined %s\n" prefix);
-  if try_invert then
+  (if try_invert then
     try
       let inv_n_string = String.concat "_" [prefix; "inv"] in
       let inv_n = Id.of_string inv_n_string in
       invert_patch inv_n env patch sigma
     with _ ->
       ()
   else
-    ()
+    ());
+  define sigma patch
 
+(* can't move "define" as the first argument to allow partial 
+   application because it messes with the type of "search" *)
+let patch_tactic env n try_invert a search sigma =
+  patch env n try_invert a search sigma 
+    (fun sigma patch ->
+      letin_pat_tac false None (Names.Name n)
+        (sigma, etrmEConstr.of_constr patch) Locusops.nowhere) 
+  
+let patch_command env n try_invert a search sigma =
+  patch env n try_invert a search sigma 
+    (fun sigma patch -> ignore (define_term n sigma patch false))
+
+  
 (* --- Commands --- *)
 
 (*
  * Command functionality for generating reusable patches
  * Patch Proof, Patch Definition, and Patch Constructor all call this
  * The latter two just pass extra guidance for now
  *)
-let patch_proof n d_old d_new cut =
+
+let patch_proof n d_old d_new cut intern patch =
   let (sigma, env) = Pfedit.get_current_context () in
-  let sigma, (old_term, new_term) = intern_defs env d_old d_new sigma in
+  let sigma, (old_term, new_term) = intern env d_old d_new sigma in
   let sigma, (d, opts) = configure env old_term new_term cut sigma in
   let change = get_change opts in
   let try_invert = not (is_conclusion change || is_hypothesis change) in
   let search _ _ = search_for_patch old_term opts d in
   patch env n try_invert () search sigma
-
+    
+let patch_proof_tactic n d_old d_new =
```

Comment:
Same here, I'd add comments explaining tactic/command version and what the difference is. They can be brief.